test/testfloat/testfloat.cpp

```diff
@@ -0,0 +1,442 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fizzy/execute.hpp>
+#include <fizzy/parser.hpp>
+#include <test/utils/floating_point_utils.hpp>
+#include <test/utils/hex.hpp>
+#include <cfenv>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+using namespace fizzy::test;
+
+namespace
+{
+/// The "usage" help message.
+///
+/// The option names match the ones from the testfloat_gen / testfloat tools.
+///
+/// TODO: Rename <float>_{ceil,floor,trunc,nearest} to <float>_roundToInt_r_<round>
+///       as in the testfloat tool?
+/// TODO: Rename <float>_trunc_to_<int> to <float>_to_<int>_r_<round>
+///       as in the testfloat tool?
+constexpr auto USAGE = R"(Fizzy TestFloat
+
+Tool for testing IEEE 754 floating-point compliance.
+Expects inputs from TestFloat's testfloat_gen.
+See http://www.jhauser.us/arithmetic/TestFloat-3/doc/TestFloat-general.html.
+
+Usage:
+  fizzy-testfloat [options] <function>
+
+  -rnear_even     Round to nearest/even [default].
+  -rminMag        Round to minimum magnitude (toward zero).
+  -rmin           Round to minimum (down).
+  -rmax           Round to maximum (up).
+  -ignoreNaNs     Do not check for specific NaN results ("is NaN" is still checked).
+
+<function>:
+  The function names match the ones from the testfloat_gen tool, with the following exceptions:
+  - <float>_to_<int> is <float>_trunc_to_<int> as the result is always rounded with -rminMag,
+  - <float>_roundToInt is <float>_{ceil,floor,trunc,nearest} as they have the rounding direction fixed.
+
+Example:
+  testfloat_gen f32_add | fizzy-testfloat f32_add
+)";
+
+// The maximum number of failures reported.
+constexpr auto max_failures = 5;
```

Comment:
Why not move this into `main` and perhaps make it configurable in the future? I'm not sure where it makes better sense, but was easier for reading `main` if it were there.

Reply:
Because it is not an option, it is easier to find it and change when at the top of the file.